modules/homepage/src/main/frontend/src/themePage/Page.jsx

```diff
@@ -0,0 +1,34 @@
+//
+//  Licensed to the Apache Software Foundation (ASF) under one
+//  or more contributor license agreements.  See the NOTICE file
+//  distributed with this work for additional information
+//  regarding copyright ownership.  The ASF licenses this file
+//  to you under the Apache License, Version 2.0 (the
+//  "License"); you may not use this file except in compliance
+//  with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing,
+//  software distributed under the License is distributed on an
+//  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+//  KIND, either express or implied.  See the License for the
+//  specific language governing permissions and limitations
+//  under the License.
+//
+
+import { useEffect } from "react";
+
+// A pseudo-component that applies the title of the given page
+function Page (props) {
+  const { children, title } = props;
+
+  // When a page is loaded, change the title of the page
+  useEffect(() => {
+    document.title = title;
+  })
```

Comment:
An empty array as a second argument will ensure the effect is only called once at the initial rendering.
```suggestion
  }, [])
```
It's not really a problem here, I checked and the effect does seem to be called only once, but it's a good practice. (Without a second argument, the effect is called every time a component (re)renders, and this can happen a lot apparently in some cases. The 2nd argument basically restricts the calls to when whatever is listed in the array changes, and an empty array as a 2nd argument ensures that the effect is only called once at the first rendering.)

Reply:
Implemented with my other changes :heavy_check_mark: